tsdb/querier/querier.go

```diff
@@ -95,11 +98,26 @@ func (q *Query) ExecWithoutEncode() QueryResponse {
 	return data
 }
 
-func (q *Query) exec(blockStream []tsdb.Block, encoding bool) interface{} {
+func (q *Query) exec(blockStream []tsdb.Block, encoding bool) (i interface{}) {
 	// start represents the starting of the timer that will calculate
 	// the total time involved in performing a particular query.
 	// This can be later benchmarked and compared with other algorithms.
 	base, stamp := getBaseResponse(q.Range)
+	defer func() {
+		var res QueryResponse
+		if q.encoding {
+			err := json.Unmarshal(i.([]byte), &res)
+			if err != nil {
+				log.Errorln(fmt.Errorf("Decoding error: %s", err.Error()).Error())
+			}
+			res.EvaluationTime = time.Since(stamp).String()
+			i, _ = json.Marshal(res)
+		} else {
+			res, _ = i.(QueryResponse)
+			res.EvaluationTime = time.Since(stamp).String()
+			i = res
+		}
+	}()
```

Comment:
This can be written as.

```suggestion
	defer func() {
		var res QueryResponse
		if !q.encoding {
		        res, ok := i.(QueryResponse)
		        if !ok {
		           log.Error("type 'QueryResponse' not received")
		        }
			res.EvaluationTime = time.Since(stamp).String()
			i = res
			return
		}
		err := json.Unmarshal(i.([]byte), &res)
		if err != nil {
			log.Errorln(fmt.Errorf("Decoding error: %s", err.Error()).Error())
		}
		res.EvaluationTime = time.Since(stamp).String()
		i, ok = json.Marshal(res)
		if !ok {
		      log.Error("invalid marshalling: ", err.Error())
	        } 
	}()
```